### PR TITLE
assistant builder Slackbot channel based routing

### PR DESCRIPTION
**Pull request by fontanierh**
Allow configuring an assistant as the default assistant on a slack channel.

This requires some painful back & forth with the `connectors` service.

**Things that are missing:**
- some kind of warning when picking a channel that is already linked with a different assistant: the behavious I am implementing is simply silently "stealing" the channel from the other assistant. It's not obvious how it should look like UI-wise. @Duncid's recommendation was to be "additive", i.e have multiple assistant reply concurrently on every slack ping, but this is currently not supported by the data model and is not supported by the slack integration.

- the "reply to all messages" mode. It makes the UI a lot more complex, and I am very dubious there is an actual use-case. The data model doesn't support it either.

Screenshot 2023-10-09 at 13 43 00

Screenshot 2023-10-09 at 13 43 07

Screenshot 2023-10-09 at 13 43 30

### CODE REVIEW

File: connectors/src/api/slack_channels_linked_with_agent.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        type: "not_found",
+        message: `Slack channel(s) not found: ${missingSlackChannelIds.join(
+          ", "
+        )}`,
+      },
+      status_code: 404,
+    });
+  }
+
+  await SlackChannel.update(
```
Comment (lasryaric) [inline connectors/src/api/slack_channels_linked_with_agent.ts:81]: We should do these two updates in a transaction. Either everything goes true or nothing, but no half state.

File: front/pages/api/w/[wId]/data_sources/[name]/managed/slack/channels_linked_with_agent.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    });
+  }
+
+  if (!auth.isBuilder()) {
+    return apiError(req, res, {
+      status_code: 403,
+      api_error: {
+        type: "data_source_auth_error",
+        message:
+          "Only the users that are `builders` for the current workspace can retrieve parents of connector resources.",
```
Comment (lasryaric) [inline front/pages/api/w/[wId]/data_sources/[name]/managed/slack/channels_linked_with_agent.ts:81]: Wrong error message?

**Review by lasryaric — CHANGES_REQUESTED** [review]

**Review by lasryaric — COMMENTED** [review]
Great job, thank you @henry, Requesting two little changes.

File: front/pages/api/w/[wId]/data_sources/[name]/managed/slack/channels_linked_with_agent.ts (hunk 2 above)
Comment (fontanierh) [reply front/pages/api/w/[wId]/data_sources/[name]/managed/slack/channels_linked_with_agent.ts:81]: Copy paste gone wrong 🙈 Thanks !

**Review by lasryaric — APPROVED** [review]

**Pull request merged**